File: android/src/com/panyam/mango/templates/parser/TokenLists.java
package com.panyam.mango.templates.parser;

/**
 * A few set token lists which are used through out the parsing stages.
 */
public class TokenLists
{
    /**
     * Expected token list for identifier or string only.
     */
    public static final TokenType COLON_OR_CLOSE_VARIABLE[] = new TokenType[] {
            TokenType.TOKEN_COLON, 
            TokenType.TOKEN_CLOSE_VARIABLE,
    };

    /**
     * Expected token list for identifier or string only.
     */
    public static final TokenType IDENT_OR_STRING[] = new TokenType[] {
            TokenType.TOKEN_IDENTIFIER, 
            TokenType.TOKEN_QUOTED_STRING,
    };

    /**
     * Tokens that can be expected when reading the variable name expression.
     */
    public static final TokenType IDENT_OR_OPEN_PAREN[] = new TokenType[] {
            TokenType.TOKEN_IDENTIFIER, 
            TokenType.TOKEN_OPEN_PAREN,
    };

    /**
     * Tokens that can be expected when reading the variable name expression.
     */
    public static final TokenType IDENT_STRING_OR_OPEN_PAREN[] = new TokenType[] {
            TokenType.TOKEN_IDENTIFIER, 
            TokenType.TOKEN_QUOTED_STRING,
            TokenType.TOKEN_OPEN_PAREN,
    };

    /**
     * Tokens that can be expected when reading the variable name expression.
     */
    public static final TokenType COMA_OR_CLOSE_PAREN[] = new TokenType[] {
            TokenType.TOKEN_COMA,
            TokenType.TOKEN_CLOSE_PAREN,
    };
    
    /**
     * Tokens that can be expected when reading the variable name expression.
     */
    public static final TokenType IDENT_TOKENS[] = new TokenType[] {
            TokenType.TOKEN_DOT, 
            TokenType.TOKEN_CLOSE_VARIABLE,
            TokenType.TOKEN_FILTER_SEPERATOR
    };
    
    /**
     * Tokens that can be expected when reading filter lists.
     */
    public static final TokenType FILTER_LIST_TOKENS[] = new TokenType[] {
            TokenType.TOKEN_CLOSE_VARIABLE,
            TokenType.TOKEN_FILTER_SEPERATOR
    };
}

File: objc/Classes/parser/TokenLists.m

#include "TokenLists.h"

/**
 * Expected token list for identifier or string only.
 */
TokenType COLON_OR_CLOSE_VARIABLE[3] = {
        TOKEN_COLON, 
        TOKEN_CLOSE_VARIABLE,
        TOKEN_UNKNOWN,
};

/**
 * Expected token list for identifier or string only.
 */
TokenType IDENT_OR_STRING[3] = {
        TOKEN_IDENTIFIER, 
        TOKEN_QUOTED_STRING,
		TOKEN_UNKNOWN,
};

/**
 * Tokens that can be expected when reading the variable name expression.
 */
TokenType IDENT_OR_OPEN_PAREN[3] = {
        TOKEN_IDENTIFIER, 
        TOKEN_OPEN_PAREN,
        TOKEN_UNKNOWN,
};

/**
 * Tokens that can be expected when reading the variable name expression.
 */
TokenType IDENT_STRING_OR_OPEN_PAREN[4] = {
        TOKEN_IDENTIFIER, 
        TOKEN_QUOTED_STRING,
        TOKEN_OPEN_PAREN,
        TOKEN_UNKNOWN,
};

/**
 * Tokens that can be expected when reading the variable name expression.
 */
TokenType COMA_OR_CLOSE_PAREN[3] = {
        TOKEN_COMA,
        TOKEN_CLOSE_PAREN,
        TOKEN_UNKNOWN,
};

/**
 * Tokens that can be expected when reading the variable name expression.
 */
TokenType IDENT_TOKENS[4] = {
        TOKEN_DOT, 
        TOKEN_CLOSE_VARIABLE,
        TOKEN_FILTER_SEPERATOR,
        TOKEN_UNKNOWN,
};

/**
 * Tokens that can be expected when reading filter lists.
 */
TokenType FILTER_LIST_TOKENS[3] = {
        TOKEN_CLOSE_VARIABLE,
        TOKEN_FILTER_SEPERATOR,
        TOKEN_UNKNOWN,
};

